### ISSUE TITLE
comptime crash when attribute takes type as ctor arg

### ISSUE THREAD

**Issue by EinScott** [beefytech/Beef#1034]
The following code crashes on compile. With the debug ide it fails an assert before where it only expects to have constant strings, not types. At least thats how far i was able to look into it. It dont know, if maybe this isnt even supposed to be possible, but right now it seems unhandled.

```
using System;

[AttributeUsage(.Class, .ReflectAttribute)]
struct AAttribute : Attribute, IComptimeTypeApply
{
	Type type;
	public this(Type type /*= typeof(float)*/)
	{
		this.type = type;
	}

	[Comptime]
	public void ApplyToType(Type type)
	{
		Compiler.EmitTypeBody(type, scope $"public const {type.ToString(.. scope .())} D = default;");
	}
}

[A(typeof(Program))] // Or something like that
class Program
{
	static void Main()
	{
		Console.WriteLine(D.ToString(.. scope .()));
		Console.Read();
	}
}
```

**Comment by bfiete** [comment]
Fixed at 617cdcbede34e69957e53dd8a3a4141ff9b24e8f

**Comment by bfiete** [comment]
Note in your code example, though, your `type.ToString` refers to the argument `type`, not the member field `type`
